lib/system/web/src/keyboard_engine/bindings.rs

```diff
@@ -0,0 +1,55 @@
+use bigint::uint::U256;
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+use crate::keyboard_engine::callback_registry::*;
+
+type BindingsMap = HashMap<U256, CallbackRegistry>;
+
+#[derive(Default)]
+pub struct Bindings {
+    pub data: RefCell<BindingsMap>,
+}
+
+impl Bindings {
+    pub fn new() -> Self{
+        Self {
+            data: RefCell::new(HashMap::new())
+        }
+    }
+
+    pub fn add(&self, key: U256, callback: Box<dyn FnMut()>) {
+        let mut data = self.data.borrow_mut();
```

Comment:
You should use Shrinkwrap here and deref coercion, so you could write `self.borrow_mut()` instead.